### DISCUSSION TITLE
How to configure 3rd part extension for python markdown

### DISCUSSION THREAD

**Discussion by Ekajto** [facelessuser/MarkdownPreview#125] [General]
Hello, 
I am a little bit a newbie and I am struggling with it. 

I am trying to use plantuml extension https://pypi.org/project/plantuml-markdown/ to generate a markdown document that will contain diagram genereted by plantuml engine. 

I have added this extension to the settings 
```
    "markdown_extensions": [
        // Python Markdown Extra with SuperFences.
        // You can't include "extra" and "superfences"
        // as "fenced_code" can not be included with "superfences",
        // so we include the pieces separately.
        "markdown.extensions.footnotes",
        "markdown.extensions.attr_list",
        "markdown.extensions.def_list",
        "markdown.extensions.tables",
        "markdown.extensions.abbr",
        //PlantUML Support added by ME
        {
        	"markdown.extensions.plantuml_markdown":
        	{
        		"server":"http://www.plantuml.com/plantuml"
        	}
        },
        "pymdownx.betterem",
        {
            "markdown.extensions.codehilite": {
                "guess_lang": false
            }
        },
```

but when I try to use it it does not work. 

I have added plantuml_markdown.py to the AppData\Roaming\Sublime Text 3\Packages\python-markdown\st3\markdown\extensions folder. 
I have downloaded required dependencies.
Whenever I try to generate md file the plantuml is not working. 

I undestand that you are not supporting 3rd extensions but my question is if I did eveyrthing correctly on MarkdownPreview side.

**Comment by facelessuser** [comment]
The plantuml extension must be available in Sublime for it to work. I don't have it as a dependency for MarkdownPreview, nor am I even certain someone has made one available for Sublime. I do not have time to maintain and update every extension available for Python Markdown in Sublime. It is enough work just backporting Python Markdown and  Pymdown Extensions (which I am the author of) to the old EOL Python 3.3 that is currently in Sublime. With that said, you can expect to see updated to Python Markdown and Pymdown Extensions (possibly with a few mild breaking changes) this week.
